src/azure-cli-tests/azure/cli/tests/preparers.py

```diff
@@ -0,0 +1,173 @@
+            pass
+
+        return request
+
+    def process_response(self, response):
+        try:
+            response['body']['string'] = self.replace(response['body']['string'])
+        except (KeyError, AttributeError):
+            pass
+
+        try:
+            response['headers']['location'] = \
+                [self.replace(l) for l in response['headers']['location']]
+        except (KeyError, AttributeError):
+            pass
+
+        return response
+
+    def replace(self, original_value):
+        return original_value.replace(self.random_name, self.moniker)
+
+
+# Resource Group Preparer and its shorthand decorator
+
+class ResourceGroupPreparer(AbstractPreparer, RecordingProcessorMixin):
+    def __init__(self, parameter_name='resource_group',
+                 parameter_name_for_location='resource_group_location', location='westus'):
+        super(ResourceGroupPreparer, self).__init__('clitest.rg.', 90)
+        self.location = location
+        self.parameter_name = parameter_name
+        self.parameter_name_for_location = parameter_name_for_location
+
+    def create_resource(self, name, **kwargs):
+        template = 'az group create --location {} --name {} --tag use=az-test'
+        execute(template.format(self.location, name))
+        return {self.parameter_name: name, self.parameter_name_for_location: self.location}
+
+    def remove_resource(self, name, **kwargs):
+        execute('az group delete --name {} --yes --no-wait'.format(name))
+
+
+# Storage Account Preparer and its shorthand decorator
+
+class StorageAccountPreparer(AbstractPreparer, RecordingProcessorMixin):
+    def __init__(self, sku='Standard_LRS', location='westus',
+                 parameter_name='storage_account',
+                 resource_group_parameter_name='resource_group', skip_delete=True):
+        super(StorageAccountPreparer, self).__init__('clitest', 24)
+        self.location = location
+        self.sku = sku
+        self.resource_group_parameter_name = resource_group_parameter_name
+        self.skip_delete = skip_delete
+        self.parameter_name = parameter_name
+
+    def create_resource(self, name, **kwargs):
+        group = self._get_resource_group(**kwargs)
+        template = 'az storage account create -n {} -g {} -l {} --sku {}'
+        execute(template.format(name, group, self.location, self.sku))
+        return {self.parameter_name: name}
+
+    def remove_resource(self, name, **kwargs):
+        group = self._get_resource_group(**kwargs)
+        execute('az storage account delete -n {} -g {} --yes'.format(name, group))
+
+    def _get_resource_group(self, **kwargs):
+        try:
+            return kwargs.get(self.resource_group_parameter_name)
+        except KeyError:
+            raise CliTestError('Storage account requires a resource group. Please decorator '
+                               'use_resource_group and add parameter {} to the test method'
```

Comment:
"Please decorator use_resource_group and add parameter {} to the test method'"

It's just that the sentence doesn't make sense.  "Please add a @ResourceGroup decorator to the test and a corresponding parameter" or something like that.